wnfs-common/src/utils/test.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{encode, BlockStore, MemoryBlockStore, CODEC_DAG_CBOR, CODEC_RAW};
+use anyhow::Result;
+use base64_serde::base64_serde_type;
+use bytes::Bytes;
+use libipld::{
+    cbor::DagCborCodec,
+    json::DagJsonCodec,
+    prelude::{Decode, Encode},
+    Cid, Ipld,
+};
+use proptest::{
+    strategy::{Strategy, ValueTree},
+    test_runner::TestRunner,
+};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{
+    collections::{BTreeMap, HashMap},
+    io::Cursor,
+};
+
+//--------------------------------------------------------------------------------------------------
+// Type Definitions
+//--------------------------------------------------------------------------------------------------
+
+type CidHandler = Box<dyn Fn(&Bytes) -> Result<Bytes>>;
+
+#[derive(Default)]
+pub struct MockStore {
+    inner: MemoryBlockStore,
+    cid_handlers: HashMap<Cid, CidHandler>,
```

Comment:
Hm. Maybe also rename this. I may have originally suggested `CidHandler` (I don't remember!), but it's not actually handling CIDs, it's handling the blocks. Perhaps we call it BlockHandler?